app/src/main/res/layout/activity_main.xml

```diff
@@ -34,6 +34,17 @@
                 app:layout_constraintStart_toStartOf="parent"
                 app:layout_constraintTop_toBottomOf="@+id/toolbar"
                 app:navGraph="@navigation/nav_graph" />
+
+            <ProgressBar
```

Comment:
maybe you want a framelayout matching parent with and height, containing inside the progressBar.. just to be able to set some opacity and disable all views while loading is shown

Reply:
I put a card instead of a frameLayout, found that it's easier to customize. Tested adding oppacity, but it looked horrible, but if it makes you happy, I'll add it